### ISSUE TITLE
First STAC document for results not easy to retrieve

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-api#397]
In the Map Viewer proposal we identified that it's not easy to share results as the "initial" STAC document returned by `GET /jobs/:id/results` is only returned for authenticated users. This way it's not easy to share that file although all children and data are already signed URLs anyway.

We should also expose the response of `GET /jobs/:id/results` as a signed URL. The idea is to simply add a Link header, which points to the file itself as a signed URL. 

For example the response for `GET /jobs/:id/results` could look like:

```http
Content-Type: application/json
Link: <https://example.openeo.org/api/jobs/2584892312376463251497856398476123746/collection.json>; rel="canonical"

{
  "$comment": "Spit out the content of the link above as a response",
  "stac_version": "1.0.0",
  "assets": {...}
}
```

Rel types for consideration: alternate, duplicate, collection/item, canonical

**Comment by m-mohr** [comment]
Alternatively, the "self" link (or "canonical") in the document could point to a signed URL. Anyway, it would be good if the results could be retrieved purely via a signed URL. I doubt that this will be hard to implement as the signed URL code is available for the other files anyway.

**Comment by m-mohr** [comment]
Added PR https://github.com/Open-EO/openeo-api/pull/400 with a proposal.

**Comment by m-mohr** [comment]
Merged, will be part of openEO API v1.2.0.
